### ISSUE TITLE
Transmit anonymous statistics via SMS

### ISSUE THREAD

**Issue by garethbowen** [medic/medic-webapp#772]
Split off from #710 

Once the twilio driver is ready, use it to submit SMS messages to the stats instance.

**Comment by abbyad** [comment]
If the twilio driver is the limiting factor I'll set up a free SMSsync gateway so that we can keep this within `Beta 2`.

**Comment by garethbowen** [comment]
The webapp side of this is now done, but it won't do anything until we have a number to send messages to.

**Comment by browndav** [comment]
Closing for acceptance testing. IIRC, the acceptance testing should just cover submission via HTTP.

**Comment by abbyad** [comment]
Actually I think this is specifically for sending via SMS, which can be configured in app_settings:

![image](https://cloud.githubusercontent.com/assets/789512/7579907/cfaa39ea-f873-11e4-854a-260d723537a5.png)

**Comment by sglangevin** [comment]
I think in order to acceptance test this, I'll need the required message format and the number for the stats.app instance (once Twilio is set up) so that I can try it out and see how it works. And if I'm going to set up in app settings to submit via SMS, how could I get the app to submit data prior to having been installed for a month?

**Comment by abbyad** [comment]
@garethbowen can you give @sglangevin some guidance on how to acceptance test this?

**Comment by garethbowen** [comment]
Sorry about taking a while to get to this @sglangevin !

Firstly, set up your own stats instance. You'll need an attached phone number and to install the states form: https://raw.githubusercontent.com/medic/medic-data/master/data/anonymous-stats/forms.json

Secondly configure your primary instance to submit stats. This can only be configured under the app_settings screen in dashboard. Enter 'sms' in the submission method and the full phone number your stats instance is attached to.

If your instance has been running more than a month you will have stats ready to submit. If the reports haven't been submitted yet then it'll automatically happen within an hour. If they have been submitted (say via http) then they won't be submitted again but we can hack them so they resubmit. Go to `/_utils/database.html?medic/_design/medic/_view/usage_stats_by_year_month` and edit at least one of the listed documents by removing the `submitted` field. Don't forget to save the document after the edit.

If however your instance hasn't been running a month you could try receiving a few reports and then winding your system clock forward into next month and waiting for an hour. You'll know it worked if you see an entry when you go to the above URL.

Sorry this is a bit complicated - let me know if you get stuck.

**Comment by garethbowen** [comment]
@sglangevin I've added a way to run schedules immediately so you don't have to wait for 3am. You'll have to update to the latest alpha, then go to:

* `/api/v1/schedule/usageStats` to generate the stats. There must be documents from the last calendar month, and no existing usageStats document for the last calendar month.
* `/api/v1/schedule/statsSubmission` to submit the stats. There must be unsubmitted usage stats documents.

We still have the problem of generating data for the last month, but at least with these you won't be waiting around for the schedule to run.

**Comment by mandric** [comment]
Is `/schedule` a bit overloaded here?  Does it have anything to do with message schedules?

**Comment by garethbowen** [comment]
@mandric Fair enough - it has nothing to do with message schedules. What about `/scheduler/`? Any other ideas?

**Comment by garethbowen** [comment]
I thought `/cron` was too technical, and hopefully one day we'll use the scheduler for more than just stats. I'll go with `/scheduler` for now.

**Comment by mandric** [comment]
Ok, my main concern was having a /schedule and a /schedules and the two doing very different things (the latter allowing us to manage messages schedules).  Probably thinking too far ahead.
